### PR TITLE
[prototype] Adjust solarize threshold on input type

### LINKED ISSUE #6840: Support all integer and floating point dtypes in prototype transform kernels?
The standard rule for dtype support for images and videos is: 

- All floating point and integer tensors are supported.
- Floating point tensors are valid in the range `[0.0, 1.0]` and integer tensors in `[0, torch.iinfo(dtype).max]` (this is currently under review since there were a few cases, where this was not true or simply not handled. See #6825)

However we have currently two kernels that only support `uint8` images or videos:

- https://github.com/pytorch/vision/blob/c84dbfad97251271a789b252a2a1a52c73f623ff/torchvision/prototype/transforms/functional/_color.py#L373-L375
- https://github.com/pytorch/vision/blob/c84dbfad97251271a789b252a2a1a52c73f623ff/torchvision/transforms/functional_tensor.py#L788-L789

This also holds for transforms v1 so this is not a problem of the new API. 

One consequence of that is that AA transforms are only supported for `uint8` images

https://github.com/pytorch/vision/blob/c84dbfad97251271a789b252a2a1a52c73f623ff/torchvision/transforms/autoaugment.py#L104-L107

since both 

https://github.com/pytorch/vision/blob/c84dbfad97251271a789b252a2a1a52c73f623ff/torchvision/transforms/autoaugment.py#L76-L77

and 

https://github.com/pytorch/vision/blob/c84dbfad97251271a789b252a2a1a52c73f623ff/torchvision/transforms/autoaugment.py#L82-L83

are used.

One possible way of mitigating this to simply have a `convert_dtype(image, torch.uint8)` in the beginning and converting back after computation. 

That is probably needed for `equalize` since we recently switched away from the histogram ops of `torch` towards our "custom" implementation to enable batch processing (#6757). However, this relies on the fact that the input is an integer and in its current form even on `uint8` due to some hardcoded constants.

For `posterize` I think it is fairly easy to provide the same functionality for float inputs directly without going through a dtype conversion first.

cc @vfdev-5 @datumbox @bjuncek

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #6840

This PR makes it possible for float images to use AutoAugmentation methods. Given that now all operators support float images, the only change necessary is to adjust the thresholds of solarize to depend on the max value of the input type (1.0 for floats, 255 for integers or PIL).

cc @vfdev-5 @bjuncek @pmeier

### CODE REVIEW

File: torchvision/prototype/transforms/_auto_augment.py [hunk 1]
```diff
@@ -137,7 +137,8 @@ def _apply_image_or_video_transform(
         elif transform_id == "Posterize":
             return F.posterize(image, bits=int(magnitude))
         elif transform_id == "Solarize":
-            return F.solarize(image, threshold=magnitude)
+            bound = 1.0 if isinstance(image, torch.Tensor) and image.is_floating_point() else 255.0
```
Comment (pmeier) [inline torchvision/prototype/transforms/_auto_augment.py:140]: We need to change this to 

```suggestion
            bound = _FT._max_value(image) isinstance(image, torch.Tensor) else 255.0
```

in #6830.

File: torchvision/prototype/transforms/_auto_augment.py [hunk 2]
```diff
@@ -505,13 +506,7 @@ def forward(self, *inputs: Any) -> Any:
...
                     aug, transform_id, magnitude, interpolation=self.interpolation, fill=self.fill
                 )
-            mix.add_(
-                # The multiplication below could become in-place provided `aug is not batch and aug.is_floating_point()`
-                # Currently we can't do this because `aug` has to be `unint8` to support ops like `equalize`.
-                # TODO: change this once all ops in `F` support floats. https://github.com/pytorch/vision/issues/6840
-                combined_weights[:, i].reshape(batch_dims)
-                * aug
-            )
+            mix.add_(combined_weights[:, i].reshape(batch_dims) * aug)
```
Comment (pmeier) [inline torchvision/prototype/transforms/_auto_augment.py:509]: Is the condition of the previous comment "aug.is_floating_point" met here?

**Review by pmeier — APPROVED** [review]
One question inline. Otherwise LGTM. Thanks Vasilis!

File: torchvision/prototype/transforms/_auto_augment.py (hunk 1 above)
Comment (datumbox) [reply torchvision/prototype/transforms/_auto_augment.py:140]: I noted this earlier at https://github.com/pytorch/vision/pull/6874#discussion_r1009457953

This is not supposed to happen yet though. Do you mean something else?

File: torchvision/prototype/transforms/_auto_augment.py (hunk 2 above)
Comment (datumbox) [reply torchvision/prototype/transforms/_auto_augment.py:509]: I don't understand the question. Aug can be float or int but it's not modified in place, so we don't care about its type. Also note there is no code change here, just removing the comment and linting.

File: torchvision/prototype/transforms/_auto_augment.py (hunk 1 above)
Comment (pmeier) [reply torchvision/prototype/transforms/_auto_augment.py:140]: Sorry, missed that. No, I mean exactly that.

File: torchvision/prototype/transforms/_auto_augment.py (hunk 2 above)
Comment (pmeier) [reply torchvision/prototype/transforms/_auto_augment.py:509]: Oh sorry, I misunderstood. So you are just removing the comment and keeping the code as is, right? So this is just doing what you proposed in https://github.com/pytorch/vision/pull/6861/files#r1008326597, right?

Comment (datumbox) [reply torchvision/prototype/transforms/_auto_augment.py:509]: Correct! All good! Just wanted to confirm you didn't mean something different. :)

**Pull request merged**